### DISCUSSION TITLE
Question about Lab2 C simulation

### DISCUSSION THREAD

**Discussion by MAGIAN1994** [bol-edu/HLS-SOC-Discussions#42] [Q&A]
想請問目前在做Lab2的C simulation時，console會報這個錯"ERROR: [SIM 211-100] 'csim_design' failed: nonzero return value."
看起來像是FIRTester的49行這邊的問題，但若不理他C synthesis照樣可以做
![擷取](https://user-images.githubusercontent.com/108878246/230433474-dc24e701-dbde-4a62-997e-dfafe74cac62.PNG)
有人有遇過類似的情況嗎

**Comment by Yanlin900414** [comment]
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145038188/ecd786e8-2574-4d16-a0bc-9d39aafd3ed5)
有，你得更改這些東西 fc -> diff 後面路徑也要改

**Comment by Josephood7** [comment]
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144367420/e6db35e2-830a-4a64-b9be-2c889e43d9e3)
如果是在Windows底下運行，前面是fc，路徑要用back slash\。
然後這裡的路徑似乎要用double back slash\\\，我也不清楚為甚麼，但我在Windows下用這樣的code寫可以work

**Comment by kevin861222** [comment]
這個問題你可以在這邊找到答案
https://github.com/bol-edu/HLS-SOC-Discussions/discussions/77#discussioncomment-7136863

**Reply by kevin861222** [reply]
另外，.dat檔案要經過dos2unix轉換，因為win系統裡換行是"\r\n" ，你要把他轉換成"\n"，否則Linux / Ubuntu 會看不懂

這邊是指令
先下載dos2unix
$ sudo apt install dos2unix

然後cd到.dat所在地
$ cd /你的目錄/

轉換
$ dos2unix out_golden.dat
